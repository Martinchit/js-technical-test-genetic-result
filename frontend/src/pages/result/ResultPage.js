import React from 'react';
import { Message } from 'semantic-ui-react';
import { InfoItem } from '../../core/components/molecules/InfoItem';
import { FlexBox } from '../../core/components/atoms/FlexBox';
import { Text } from '../../core/components/atoms/Text';
import { LoadingSpinner } from '../../core/components/molecules/LoadingSpinner';
import { useFetchGeneticResult } from '../../core/lib/hooks/useFetchGeneticResult';

export const ResultPage = () => {
  const [result, error, loading] = useFetchGeneticResult();

  return (
    <FlexBox $elevation padding="5%" flexDirection="column" backgroundColor="white">
      <Text fontSize={1.8} fontColor="black">
        Result
      </Text>
      <FlexBox flexDirection="column" margin="10px 0">
        {loading ? (
          <LoadingSpinner inverted={true} />
        ) : !error ? (
          <React.Fragment>
            {Object.entries(result.data).map(([k, v]) => (
              <InfoItem key={`result_${k}`} header={k} description={v} />
            ))}
          </React.Fragment>
        ) : (
          <Message error header="Fetch Error" content="Please refresh" />
        )}
      </FlexBox>
    </FlexBox>
  );
};
