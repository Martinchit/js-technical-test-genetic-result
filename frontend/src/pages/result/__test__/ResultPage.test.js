import React from "react";
import { shallow } from 'enzyme';
import * as redux from 'react-redux';

import { ResultPage } from "../ResultPage";

describe("ResultPage", () => {
  let spyOnUseSelector;
  let spyOnUseDispatch;
  let mockDispatch;
  const mockState = {
    auth: {
      token: 'test',
      loggedIn: true
    }
  };

  beforeEach(() => {
    spyOnUseSelector = jest.spyOn(redux, 'useSelector').mockImplementation(cb => cb(mockState));
    spyOnUseDispatch = jest.spyOn(redux, 'useDispatch');
    mockDispatch = jest.fn();
    spyOnUseDispatch.mockReturnValue(mockDispatch);
  })

  afterEach(() => {
    jest.restoreAllMocks();
  })

  it("renders correctly", () => {
    const wrapper = shallow(<ResultPage />)
    expect(wrapper).toMatchSnapshot()
  })
})