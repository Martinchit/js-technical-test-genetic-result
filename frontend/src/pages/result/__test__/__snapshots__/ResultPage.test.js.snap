// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPage renders correctly 1`] = `ShallowWrapper {}`;
